src/share/parallel.c

```diff
@@ -83,4 +83,30 @@ compress_finish_par(zfp_stream* stream, bitstream** src, uint chunks)
     stream_wseek(dst, offset);
 }
 
-#endif
+/* initialize per-thread bit streams for parallel decompression */
+static bitstream**
+decompress_init_par(zfp_stream* stream, const zfp_field* field, uint chunks)
+{
+  uint i;
+  void * buffer = stream_begin(zfp_stream_bit_stream(stream));
+  bitstream** bs = (bitstream**)malloc(chunks * sizeof(bitstream*));
```

Comment:
What happens if `malloc` fails? You are not checking if `bs` is not `NULL`

Reply:
Good point.  I believe this is the only place in zfp where malloc is not being checked.

Reply:
This malloc is taken from line 51 of compress_init_par, where it is also unchecked. I will include a check for this in my function.
I will also raise an issue for the unchecked mallocs in compress_init_par.